src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -93,6 +97,7 @@ export class ExamParticipationComponent implements OnInit, OnDestroy {
             this.examId = parseInt(params['examId'], 10);
             this.examParticipationService.loadExam(this.courseId, this.examId).subscribe((exam) => {
                 this.exam = exam;
+                this.individualStudentEndDate = exam.endDate ? exam.endDate : this.serverDateService.now();
```

Comment:
WOW, does this default value make a lot of sense? Should this rather be `null` and handled appropriately?

Reply:
Well, this and the case in your other comment are edge cases which shouldn't (can't) happen. If there is no end / start date in the exam something has to go really wrong, since it is also prevented when creating / updating an exam.
This default case is the same as passing `null` to `moment()`, just handled with the server-sync-time, ultimately setting the exam to _over_ if this happens.